siliconcompiler/core.py

```diff
@@ -4140,10 +4140,32 @@ def run(self):
                     # TODO: breakpoint logic:
                     # if task is breakpoint, then don't launch while len(running_tasks) > 0
 
+                    successful_deps = []
+                    deps_full_len = len(deps)
+                    tool, _ = self._get_tool_task(
+                        processes[task]._args[0],
+                        processes[task]._args[1])
+
                     # Clear any tasks that have finished from dependency list.
                     for in_task in deps.copy():
                         if status[in_task] != TaskStatus.PENDING:
                             deps.remove(in_task)
+                        if status[in_task] == TaskStatus.SUCCESS:
+                            successful_deps.append(in_task)
+                        if status[in_task] == TaskStatus.ERROR:
+                            # Fail if any dependency failed for non-builtin task
+                            if not self._is_builtin(tool, task):
```

Comment:
Watch out when using `self._is_builtin(...)` - based on other uses, it expects the `task` argument to be a task name as returned by `_get_tool_task()`, but the `task` you're passing in is what we'd probably call a "node" or "node name" now (a step name + index concatenated).

It works regardless since `_is_builtin()` ignores the `task` argument for now, but it would be good to make the usage match what's expected in case that changes in the future. Part of this confusion is also related to the fact that we changed the meaning of the term "task" at some point, so it would probably be good to refactor the other usage of "task" to be "node" in a future PR.

Reply:
Ok changed that plus the meaning of node and task around it.

For the full refactor there is this ticket SC-586